### DISCUSSION TITLE
Avoiding repetitive code

### DISCUSSION THREAD

**Discussion by cristw** [uNetworking/uWebSockets.js#407] [Q&A] [answered]
The code below is to send the main.js file from the server on request. And I have to write something like that for every single file that the client requests. Is there a way to do it more efficiently?
The client can request any file from the /public folder. Is there a way to put the request link and add it directly with the _dirname without doing it all manually for each file?
Even for files that are not .js (for example: model.glb)

```javascript
uwss.get('/public/js/main.js', (res, req) => {
    var data = fs.readFileSync(__dirname + '/public/js/main.js');
    if(data){
        res.write(data);
    }
    res.end();
});
```

**Comment by ghost** [comment, marked as answer]
In the user manual and in examples you can see the use of parameter routes.

/public/js/:filename

Where you can do

let filename = req.getParameter(0);

**Comment by cristw** [comment]
Thanks!

But what if I want to put all the parameters? Potentially allowing all folders and files in the /public folder to be accessible.

All I can think is:

```javascript
uwss.get('/public/:a/:b', (res, req) => {
    var data = fs.readFileSync(__dirname + '/public/' + req.getParameter(0) + '/' + req.getParameter(1) + '/' + req.getParameter(2))...etc. ;
    if(data){
        res.write(data);
    }
    res.end();
});
```

**Comment by hst-m** [comment]
you dont want to use readFileSync though you want the async version readFile

```javascript
app.get('/public/*',(res,req)=>{
   res.onAborted(()=>res.ab=true)
   fs.readFile(__dirname+req.getUrl(),(err,data)=>{
      if(res.ab)return
      if(err)return res.writeStatus('404').end()
      res.end(data)
   })
})
```
I dont use readFile in my apps per request, I load files once into JS memory on server start and then watch them for changes, but I dont have a huge amount of files. if you have large files use fs.createReadStream

**Reply by cristw** [reply]
Wow! It worked flawlessly. Thanks for pointing out the readFileSync thing, I don't know what was I thinking.

What does this do?

```javascript
res.onAborted(()=>res.ab=true)
```

**Reply by hst-m** [reply]
that is required if you do anything async so you dont try to respond to a closed connection when code resumes
